drivers/include/periph/adc.h

```diff
@@ -30,6 +30,18 @@
  * waiting for the result of a conversion (e.g. through putting the calling
  * thread to sleep while waiting for the conversion results).
  *
+ * @section     sec_pm (Low-) power implications
+ *
+ * The ADC peripheral(s) **should** only be powered on while adc_read() is
+ * active. For implementing adc_read() this means, that the peripheral should be
+ * powered on (i.e. through peripheral clock gating) at the beginning of the
+ * function and it should be powered back off at the end of the function.
+ *
```

Comment:
If having the adc peripheral "powered" costs too much energy, wouldn't explicit poweron()/poweroff() make more sense?